src/middlewared/middlewared/plugins/webui/enclosure.py

```diff
@@ -0,0 +1,294 @@
+from middlewared.schema import accepts, returns, List, Dict
+from middlewared.service import Service
+
+
+class WebUIEnclosureService(Service):
+    class Config:
+        namespace = 'webui.enclosure'
+        private = True
+        cli_private = True
+
+    def dashboard_impl(self):
+        disks_to_pools = dict()
+        enclosures = self.middleware.call_sync('enclosure2.query')
+        if enclosures:
+            disks_to_pools = self.middleware.call_sync('zpool.status', {'real_paths': True})
+            for enc in enclosures:
+                for disk_slot, slot_info in enc['elements']['Array Device Slot'].items():
+                    # remove some values that webUI doesn't use
+                    slot_info.pop('original')
+                    slot_info.pop('value')
+                    slot_info.pop('value_raw')
+                    pool_info = None
+                    if slot_info['dev'] and (pool_info := disks_to_pools['disks'].get(slot_info['dev'])):
+                        index = pool_info['vdev_disks'].index(slot_info['dev'])
+                        pool_info['vdev_disks'][index] = {
+                            'enclosure_id': enc['id'],
+                            'slot': int(disk_slot),
+                            'dev': slot_info['dev'],
+                        }
+                    slot_info.update({'pool_info': pool_info})
+
+        return enclosures
+
+    @accepts()
+    @returns(List('enclosures', items=[Dict('enclosure', additional_attrs=True)], example=[
+        {
+            "name": "iX 4024Sp e001",
+            "model": "M60",
+            "controller": True,
+            "dmi": "TRUENAS-M60-HA",
+            "status": [
+                "OK"
+            ],
+            "id": "5b0bd6d1a309b57f",
+            "vendor": "iX",
+            "product": "4024Sp",
+            "revision": "e001",
+            "bsg": "/dev/bsg/17:0:19:0",
+            "sg": "/dev/sg433",
+            "pci": "17:0:19:0",
+            "rackmount": True,
+            "top_loaded": False,
+            "front_slots": 24,
+            "rear_slots": 4,
+            "internal_slots": 0,
+            "elements": {
+                "Array Device Slot": {
+                    "1": {
+                        "descriptor": "slot00",
```

Comment:
Do we need so many items in the example?

Reply:
Nope, fixed.